jcli/src/jcli_app/certificate/new_vote_cast.rs

```diff
@@ -0,0 +1,38 @@
+use crate::jcli_app::certificate::{write_cert, Error};
+use chain_impl_mockchain::{
+    certificate::{Certificate, VoteCast, VotePlanId},
+    vote::{Choice, Payload},
+};
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+/// create a vote cast certificate
+#[derive(StructOpt)]
+pub struct VoteCastCmd {
+    /// the vote plan identified on the blockchain
+    pub vote_plan_id: VotePlanId,
+
+    /// the number of proposal in the vote plan you vote for
+    pub proposal_index: u8,
+
+    /// the number of choice within the proposal you vote for
+    pub choice: u8,
+
+    /// write the output to the given file or print it to the standard output if not defined
+    #[structopt(long = "output")]
+    pub output: Option<PathBuf>,
+}
+
+impl VoteCastCmd {
+    pub fn exec(self) -> Result<(), Error> {
+        let vote_cast = VoteCast::new(
+            self.vote_plan_id,
+            self.proposal_index,
+            Payload::Public {
+                choice: Choice::new(self.choice),
+            },
```

Comment:
add an option `--public` so we already enforce the users to be aware that the vote is disclosed on chain

Reply:
if the `--public` is not set, then the command fail, because the crypto for the privacy preserving part is not finalised yet.

Reply:
done